lang/ruby/test/test_schema.rb

```diff
@@ -160,4 +160,101 @@ def test_to_avro_handles_falsey_defaults
       ]
     }
   end
+
+  def test_support_for_doc_attribute
+    # Testing doc support for fields
+    field_schema_json = Avro::Schema.parse <<-SCHEMA
+      {
+        "type": "record",
+        "name": "Record",
+        "namespace": "my.name.space",
+        "fields": [
+          {
+            "name": "name",
+            "type": "boolean",
+            "doc": "documentation"
+          }
+        ]
+      }
+    SCHEMA
+
+    field_schema_hash =
+      {
+        'type' => 'record',
+        'name' => 'Record',
+        'namespace' => 'my.name.space',
+        'fields' => [
+          {
+            'name' => 'name',
+            'type' => 'boolean',
+            'doc' => 'documentation'
+          }
+        ]
+      }
+
+    assert_equal field_schema_hash, field_schema_json.to_avro
+
+    # Testing doc support for records
```

Comment:
I'd make these separate test methods: `test_field_doc_attribute`, `test_record_doc_attribute`, etc. I don't think it adds any value to combine them. If one breaks, then they're different areas of the code.